GPU/GPUTracking/Global/GPUChainTracking.cxx

```diff
@@ -803,16 +805,23 @@ int GPUChainTracking::RunTPCClusterizer()
       unsigned int iSlice = iSliceBase + lane;
       GPUTPCClusterFinder& clusterer = processors()->tpcClusterer[iSlice];
       GPUTPCClusterFinder& clustererShadow = doGPU ? processorsShadow()->tpcClusterer[iSlice] : clusterer;
+      SetupGPUProcessor(&clusterer, false);
+      clusterer.mPmemory->counters.nPeaks = clusterer.mPmemory->counters.nClusters = 0;
 
+      clusterer.PrepareMC();
```

Comment:
Can't we just move the `PrepareMC()` inside the `if (propagateMClabels)`. Then we can also remove all the asserts, since they will be automatically true.